bundles/action/org.openhab.action.pushover/src/main/java/org/openhab/action/pushover/internal/Pushover.java

```diff
@@ -312,33 +340,63 @@ public static boolean pushover(@ParamDoc(name = "apiKey", text = "Your applicati
-                addEncodedParameter(data, MESSAGE_KEY_SOUND, sound);
+                parts.add(new StringPart(MESSAGE_KEY_SOUND, sound));
             } else if (!StringUtils.isEmpty(defaultSound)) {
-                addEncodedParameter(data, MESSAGE_KEY_SOUND, defaultSound);
+                parts.add(new StringPart(MESSAGE_KEY_SOUND, defaultSound));
             }
 
             if (isValueInList(API_HIGH_PRIORITY_LIST, priority)) {
                 if (retry >= API_MIN_RETRY_SECONDS) {
-                    addEncodedParameter(data, MESSAGE_KEY_RETRY, String.valueOf(retry));
+                    parts.add(new StringPart(MESSAGE_KEY_RETRY, String.valueOf(retry)));
                 } else {
                     logger.warn("Retry value of {} is too small. Using default value of {}.", retry,
                             API_MIN_RETRY_SECONDS);
-                    addEncodedParameter(data, MESSAGE_KEY_RETRY, String.valueOf(API_MIN_RETRY_SECONDS));
+                    parts.add(new StringPart(MESSAGE_KEY_RETRY, String.valueOf(API_MIN_RETRY_SECONDS)));
                 }
 
                 if (expire <= API_MAX_EXPIRE_SECONDS) {
-                    addEncodedParameter(data, MESSAGE_KEY_EXPIRE, String.valueOf(expire));
+                    parts.add(new StringPart(MESSAGE_KEY_EXPIRE, String.valueOf(expire)));
                 } else {
                     logger.warn("Expire value of {} is too large. Using default value of {}.", expire,
                             API_MAX_EXPIRE_SECONDS);
-                    addEncodedParameter(data, MESSAGE_KEY_EXPIRE, String.valueOf(API_MAX_EXPIRE_SECONDS));
+                    parts.add(new StringPart(MESSAGE_KEY_EXPIRE, String.valueOf(API_MAX_EXPIRE_SECONDS)));
                 }
             }
+            if (!StringUtils.isEmpty(attachment)) {
+                logger.debug("Push the image attachment '{}'.", attachment);
+                try {
+                    File f = new File(attachment);
+                    FilePart fp = new FilePart(MESSAGE_KEY_ATTACHMENT, f.getName(), f);
+                    fp.setContentType("image/jpeg");
+                    parts.add(fp);
+                } catch (IOException | RuntimeException e) {
+                    logger.warn("Could not process file '{}': {}.", attachment, e.getMessage());
+                }
+            }
+
+            PostMethod httpPost = new PostMethod(API_URL);
+
+            httpPost.setRequestEntity(
+                    new MultipartRequestEntity(parts.toArray(new Part[parts.size()]), httpPost.getParams()));
 
-            String content = data.toString();
-            logger.debug("Executing post to " + API_URL + " with the following content: " + content);
-            String response = HttpUtil.executeUrl("POST", API_URL, IOUtils.toInputStream(content), CONTENT_TYPE,
-                    timeout);
+            logger.debug("Executing post to " + API_URL + " with the following content: " + httpPost);
```

Comment:
This needs to use parameterized logging instead of string concatenation.

Reply:
done